src/mon/OSDMonitor.cc

```diff
@@ -7688,31 +7688,37 @@ int OSDMonitor::get_crush_rule(const string &rule_name,
   return 0;
 }
 
-int OSDMonitor::check_pg_num(int64_t pool, int pg_num, int size, ostream *ss)
+int OSDMonitor::check_pg_num(int64_t pool, int pg_num, int size, int crush_rule, ostream *ss)
 {
   auto max_pgs_per_osd = g_conf().get_val<uint64_t>("mon_max_pg_per_osd");
-  auto num_osds = std::max(osdmap.get_num_in_osds(), 3u);   // assume min cluster size 3
-  auto max_pgs = max_pgs_per_osd * num_osds;
   uint64_t projected = 0;
-  if (pool < 0) {
-    projected += pg_num * size;
-  }
-  for (const auto& i : osdmap.get_pools()) {
-    if (i.first == pool) {
-      projected += pg_num * size;
-    } else {
-      projected += i.second.get_pg_num_target() * i.second.get_size();
-    }
-  }
+  unsigned osd_num = 0;
+  CrushWrapper newcrush;
+  _get_pending_crush(newcrush);
+  set<int> roots;
+  newcrush.find_takes_by_rule(crush_rule, &roots);
+  for (auto root : roots) {
+    const char *rootname = newcrush.get_item_name(root);
+    set<int> osd_ids;
+    newcrush.get_leaves(rootname, &osd_ids);
+    for (auto &id : osd_ids){
+      uint64_t num_pgs = mapping.get_osd_acting_pgs(id).size();
+      projected += num_pgs;
+      osd_num ++;
```

Comment:
could you remove the space after `osd_num`? or better off removing this line. and put
```c++
osd_num += osd_ids.size();
```
out of the loop.